### ISSUE TITLE
Accelerate not found when clicking train

### ISSUE THREAD

**Issue by nanafy** [devilismyfriend/StableTuner#24]
<img width="1275" alt="Capture" src="https://user-images.githubusercontent.com/19253644/208249673-b5e9798e-4dfe-4a25-8455-35bb0ef803d0.PNG">

1. Tried fresh install 3 times. Tried miniconda and anaconda fresh install. 
2. Tried also adding anaconda to the path upon installation, still did not work
3. I also have python 3.10 installed, not sure if that has anything to do with it. 

Anyone with more experience that has any ideas, I'd gladly appreciate it.

**Comment by nanafy** [comment]
Womp

**Comment by devilismyfriend** [comment]
This an issue with your system's PATH variables

**Comment by nanafy** [comment]
Ty! I hate PATH issues they are messy. What do you think needs to be properly in my path, is it a python or conda pathing issue?
Going to try and give it another stab right now.

**Comment by nanafy** [comment]
`
# packages in environment at C:\Users\basil\.conda\envs\ST:
#
# Name                    Version                   Build  Channel
absl-py                   1.3.0                    pypi_0    pypi
albumentations            1.3.0                    pypi_0    pypi
antlr4-python3-runtime    4.9.3                    pypi_0    pypi
aom                       3.5.0                h63175ca_0    conda-forge
bitsandbytes              0.35.0                   pypi_0    pypi
blosc                     1.21.3               hdccc3a2_0    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
c-blosc2                  2.6.1                h183a6f4_0    conda-forge
ca-certificates           2022.12.7            h5b45459_0    conda-forge
cachetools                5.2.0                    pypi_0    pypi
certifi                   2022.12.7                pypi_0    pypi
cfitsio                   4.2.0                h9ebe7e4_0    conda-forge
charls                    2.3.4                h39d44d4_0    conda-forge
charset-normalizer        2.1.1                    pypi_0    pypi
click                     8.1.3           win_pyhd8ed1ab_2    conda-forge
cloudpickle               2.2.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
cudatoolkit               11.8.0              h09e9e62_11    conda-forge
customtkinter             5.0.3                    pypi_0    pypi
cytoolz                   0.12.0          py310h8d17308_1    conda-forge
darkdetect                0.8.0                    pypi_0    pypi
dask-core                 2022.12.1          pyhd8ed1ab_0    conda-forge
dav1d                     1.0.0                h8ffe710_1    conda-forge
diffusers                 0.10.2                   pypi_0    pypi
einops                    0.6.0                    pypi_0    pypi
fairscale                 0.4.13                   pypi_0    pypi
filelock                  3.8.2                    pypi_0    pypi
freetype                  2.12.1               h546665d_1    conda-forge
fsspec                    2022.11.0          pyhd8ed1ab_0    conda-forge
ftfy                      6.1.1                    pypi_0    pypi
giflib                    5.2.1                h8d14728_2    conda-forge
git                       2.39.0               h57928b3_0    conda-forge
gitdb                     4.0.10                   pypi_0    pypi
gitpython                 3.1.29                   pypi_0    pypi
google-auth               2.15.0                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
grpcio                    1.51.1                   pypi_0    pypi
huggingface-hub           0.11.1                   pypi_0    pypi
idna                      3.4                      pypi_0    pypi
imagecodecs               2022.9.26       py310hcad817f_5    conda-forge
imageio                   2.23.0             pyhfa7a67d_0    conda-forge
importlib-metadata        5.2.0                    pypi_0    pypi
intel-openmp              2023.0.0         h57928b3_25922    conda-forge
joblib                    1.2.0                    pypi_0    pypi
jpeg                      9e                   h8ffe710_2    conda-forge
jxrlib                    1.1                  h8ffe710_2    conda-forge
krb5                      1.20.1               heb0366b_0    conda-forge
lcms2                     2.14                 ha5c8aab_1    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libaec                    1.0.6                h39d44d4_0    conda-forge
libavif                   0.11.1               h8f5c5bc_0    conda-forge
libblas                   3.9.0              16_win64_mkl    conda-forge
libbrotlicommon           1.0.9                hcfcfb64_8    conda-forge
libbrotlidec              1.0.9                hcfcfb64_8    conda-forge
libbrotlienc              1.0.9                hcfcfb64_8    conda-forge
libcblas                  3.9.0              16_win64_mkl    conda-forge
libcurl                   7.86.0               h68f0423_2    conda-forge
libdeflate                1.14                 hcfcfb64_0    conda-forge
libffi                    3.4.2                h8ffe710_5    conda-forge
libhwloc                  2.8.0                h039e092_1    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
liblapack                 3.9.0              16_win64_mkl    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libsqlite                 3.40.0               hcfcfb64_0    conda-forge
libssh2                   1.10.0               h9a1e1f7_3    conda-forge
libtiff                   4.5.0                hc4f729c_0    conda-forge
libwebp-base              1.2.4                h8ffe710_0    conda-forge
libxcb                    1.13              hcd874cb_1004    conda-forge
libxml2                   2.10.3               hc3477c8_0    conda-forge
libzlib                   1.2.13               hcfcfb64_4    conda-forge
libzopfli                 1.0.3                h0e60522_0    conda-forge
lightning-utilities       0.4.2                    pypi_0    pypi
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lz4-c                     1.9.3                h8ffe710_1    conda-forge
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markdown                  3.4.1                    pypi_0    pypi
mkl                       2022.1.0           h6a75c08_874    conda-forge
msys2-conda-epoch         20160418                      1    conda-forge
networkx                  2.8.8              pyhd8ed1ab_0    conda-forge
nodejs                    18.12.1              h57928b3_0    conda-forge
numpy                     1.24.0          py310h4a8f9c9_0    conda-forge
oauthlib                  3.2.2                    pypi_0    pypi
omegaconf                 2.3.0                    pypi_0    pypi
openjpeg                  2.5.0                ha2aaf27_2    conda-forge
openssl                   3.0.7                hcfcfb64_1    conda-forge
packaging                 22.0               pyhd8ed1ab_0    conda-forge
partd                     1.3.0              pyhd8ed1ab_0    conda-forge
pillow                    9.3.0                    pypi_0    pypi
pip                       22.3.1             pyhd8ed1ab_0    conda-forge
protobuf                  3.20.1                   pypi_0    pypi
pthread-stubs             0.4               hcd874cb_1001    conda-forge
pthreads-win32            2.9.1                hfa6e2cd_3    conda-forge
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pyperclip                 1.8.2                    pypi_0    pypi
python                    3.10.0          hcf16a7b_3_cpython    conda-forge
python_abi                3.10                    3_cp310    conda-forge
pytorch-lightning         1.8.5.post0              pypi_0    pypi
pywavelets                1.3.0           py310h9b08ddd_2    conda-forge
pyyaml                    6.0             py310h8d17308_5    conda-forge
qudida                    0.0.4                    pypi_0    pypi
regex                     2022.10.31               pypi_0    pypi
requests                  2.28.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
safetensors               0.2.6                    pypi_0    pypi
scikit-image              0.19.3          py310h1c4a608_2    conda-forge
scikit-learn              1.2.0                    pypi_0    pypi
scipy                     1.9.3           py310h578b7cb_2    conda-forge
setuptools                65.6.3             pyhd8ed1ab_0    conda-forge
smmap                     5.0.0                    pypi_0    pypi
snappy                    1.1.9                hfb803bf_2    conda-forge
sqlite                    3.40.0               hcfcfb64_0    conda-forge
tbb                       2021.7.0             h91493d7_1    conda-forge
tensorboard               2.11.0                   pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorboardx              2.5.1                    pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2022.10.10         pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h8ffe710_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
torch                     1.12.1+cu116             pypi_0    pypi
torchmetrics              0.11.0                   pypi_0    pypi
tqdm                      4.64.1                   pypi_0    pypi
typing-extensions         4.4.0                    pypi_0    pypi
tzdata                    2022g                h191b570_0    conda-forge
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
urllib3                   1.26.13                  pypi_0    pypi
vc                        14.3                 h3d8a991_9    conda-forge
vs2015_runtime            14.32.31332          h1d6e394_9    conda-forge
wcwidth                   0.2.5                    pypi_0    pypi
wheel                     0.38.4             pyhd8ed1ab_0    conda-forge
xformers                  0.0.14.dev0              pypi_0    pypi
xorg-libxau               1.0.9                hcd874cb_0    conda-forge
xorg-libxdmcp             1.1.3                hcd874cb_0    conda-forge
xz                        5.2.6                h8d14728_0    conda-forge
yaml                      0.2.5                h8ffe710_2    conda-forge
yarn                      1.22.19              h57928b3_0    conda-forge
zfp                       1.0.0                h63175ca_3    conda-forge
zipp                      3.11.0                   pypi_0    pypi
zlib-ng                   2.0.6                h8ffe710_0    conda-forge
zstd                      1.5.2                h7755175_4    conda-forge
`

**Comment by nanafy** [comment]
I can see accelerators is just not installing lol

**Comment by nanafy** [comment]
Ok I am pretty close to figuring it out. It seems somehow my installation of Auto111 is conflicting with where accelerate is located. It thinks its already installed and wont actually download it into the ST env.. What a mess 

(ST) ...\AStableTuner\StableTuner>pip install accelerate
Requirement already satisfied: accelerate in c:\users\xxxx\appdata\roaming\python\python310\site-packages (0.15.0)
Requirement already satisfied: packaging>=20.0 in c:\users\xxx\.conda\envs\st\lib\site-packages (from accelerate) (22.0)
Requirement already satisfied: numpy>=1.17 in c:\users\xxx\.conda\envs\st\lib\site-packages (from accelerate) (1.24.0)
Requirement already satisfied: torch>=1.4.0 in c:\users\xxx\.conda\envs\st\lib\site-packages (from accelerate) (1.12.1+cu116)
Requirement already satisfied: psutil in c:\users\xxxx\appdata\roaming\python\python310\site-packages (from accelerate) (5.9.4)
Requirement already satisfied: pyyaml in c:\users\xxx\.conda\envs\st\lib\site-packages (from accelerate) (6.0)
Requirement already satisfied: typing-extensions in c:\users\xxx\.conda\envs\st\lib\site-packages (from torch>=1.4.0->accelerate) (4.4.0)

**Comment by nanafy** [comment]
Got it to work, you can mark it closed. F****** finally. Third times the charm. Had no time. Thanks a lot. Excited to try this out tomorrow. 

For anyone else who has this issue:
- Delete Python site-packages
- Remove all instances of python from both local and global PATH variables
- Uninstall Python and reinstall it only to local , Make sure your python is contained in a local path and its not global
- If python not found, make sure its at the top of your path list along with scripts to override the windows/local/app path crap
- Reinstall program in question

**Comment by devilismyfriend** [comment]
Happy to hear you got it working!

**Comment by ZeroCool22** [comment]
So it will mess the AUTO installation?

Why it have conflict with it?

 it should install everything in a isolated path like AUTO's does.

**Comment by devilismyfriend** [comment]
It does.

**Comment by nanafy** [comment]
> So it will mess the AUTO installation?
> 
> Why it have conflict with it?
> 
> it should install everything in a isolated path like AUTO's does.

It does like the author said. My installation was a mess. You wont run into my issue. I had a lot of things installed globally for different reasons that suit my personal needs.
